zio-kafka/src/main/scala/zio/kafka/consumer/ConsumerSettings.scala

```diff
@@ -21,15 +21,16 @@ import zio.kafka.security.KafkaCredentialStore
  *   subscribed for long periods during its lifetime, this timeout should take that into account as well. When the
  *   timeout expires, the plainStream/partitionedStream/etc will fail with a [[Consumer.RunloopTimeout]].
  */
-case class ConsumerSettings(
+final case class ConsumerSettings(
   bootstrapServers: List[String],
   properties: Map[String, AnyRef],
   closeTimeout: Duration,
   pollTimeout: Duration,
   offsetRetrieval: OffsetRetrieval = OffsetRetrieval.Auto(),
   rebalanceListener: RebalanceListener = RebalanceListener.noop,
   restartStreamOnRebalancing: Boolean = false,
-  runloopTimeout: Duration = ConsumerSettings.defaultRunloopTimeout
+  runloopTimeout: Duration = ConsumerSettings.defaultRunloopTimeout,
+  disableOptimisticResume: Boolean = false // By default, the "optimistic resume" optimisation is enabled
```

Comment:
I prefer positive wording to prevent double negations:

```suggestion
  enableOptimisticResume: Boolean = true
```

or better yet:

```suggestion
  optimisticResume: Boolean = true
```

Reply:
Changed. Thanks!